test/cpp/api/autograd.cpp

```diff
@@ -0,0 +1,129 @@
+#include <gtest/gtest.h>
+
+#include <torch/autograd.h>
+
+#include <torch/utils.h>
+#include <test/cpp/api/support.h>
+
+using namespace torch;
+
+#define ASSERT_VARIABLE_EQ(a,b) ASSERT_TRUE(torch::allclose((a),(b)))
+
+std::string graph_desc(std::shared_ptr<autograd::Node> node) {
+  if (!node) {
+    return "None";
+  }
+  auto result = node->name() + "(";
+  auto next_edges = node->next_edges();
+  for(auto& edge : next_edges) {
+    result += graph_desc(edge.function);
+  }
+  return result+")";
+}
+
+TEST(CustomAutogradTest, CustomFunction) {
+  struct MyFunction : public torch::autograd::Function<MyFunction> {
+    static variable_list forward(AutogradContext *ctx, Variable var1, int mul, Variable var2) {
+      ctx->saved_data["mul"] = mul;
```

Comment:
Didn't that change to `save`?

Reply:
It changed from `save` to `saved_data`. I'll update the comment.

Reply:
One quick thought: I'm not looking forward to someone trying to save a Variable in `ctx->saved_data`. Does it error if you try to do this? Can we enforce that people not do this?

Reply:
It does not enforce that only non variables are saved in saved_data, because the python API has the same behavior.